Source/WebKit/WebProcess/glib/UserMediaCaptureManager.cpp

```diff
@@ -67,23 +67,20 @@ void UserMediaCaptureManager::validateUserMediaRequestConstraints(WebCore::Media
 void UserMediaCaptureManager::getMediaStreamDevices(bool revealIdsAndLabels, GetMediaStreamDevicesCallback&& completionHandler)
 {
     RealtimeMediaSourceCenter::singleton().getMediaStreamDevices([completionHandler = WTFMove(completionHandler), revealIdsAndLabels](auto&& devices) mutable {
-        Vector<CaptureDeviceWithCapabilities> devicesWithCapabilities;
-
-        devicesWithCapabilities.reserveInitialCapacity(devices.size());
-        for (auto& device : devices) {
+        auto devicesWithCapabilities = WTF::compactMap(devices, [&](auto& device) -> std::optional<CaptureDeviceWithCapabilities> {
```

Comment:
Is this returning a `Vector< CaptureDeviceWithCapabilities>`? Or is it now a `Vector<std::optional< CaptureDeviceWithCapabilities>>`?

Reply:
It is returning a `Vector<CaptureDeviceWithCapabilities>` since I am using WTF::compactMap(), compactMap() extracts the optional value and discards the nullopt ones.